plugins/fileservices/fileservices.cpp

```diff
@@ -3140,3 +3140,56 @@ FILESERVICES_API void FILESERVICES_CALL fsClearExpireDays(ICodeContext * ctx, co
     else
         throw makeStringExceptionV(0, "ClearExpireDays: Could not find logical file %s", lfn.str());
 }
+
+bool getDefaultValue(const char * processName, const char * propertyName, StringBuffer & strDefaultValue)
+{
+    // Returns with true if a default value retrieved and false if not.
+
+    #ifdef _CONTAINERIZED
+        // TBD
+
+    #else
+        Owned<IConstEnvironment> daliEnv = openDaliEnvironment();
+        Owned<IPropertyTree> env = getEnvironmentTree(daliEnv);
+
+        if (env.get())
+        {
+            Owned<IPropertyTreeIterator> processIter = env->getElements(processName);
+            if (processIter->first())
+            {
+                if (processIter->query().hasProp(propertyName))
+                {
+                    processIter->query().getProp(propertyName, strDefaultValue);
+                    return true;
+                }
+            }
+        }
+    #endif
+    return false;
+}
+
+FILESERVICES_API bool FILESERVICES_CALL fsGetNoCommonDefault()
+{
+    // This function returns the value of the DFUServerProcess' 'noCommon' property, if it is exists. 
+    // Othervise it returns with true, the recent default value of 'noCommon'
+
+    StringBuffer strDefaultValue;
+    if (getDefaultValue("Software/DfuServerProcess", "@noCommon", strDefaultValue))
```

Comment:
this could be called quite often and it's not that cheap (openDaliEnvironment() etc.).
To avoid the cost per call, you could use a singleton pattern to ensure the action is only done on 1st call only.